### ISSUE TITLE
Circonus adapter doesn't support dimensions

### ISSUE THREAD

**Issue by geeknoid** [istio/istio#3381]
The Circonus adapter isn't currently looking at a metric's dimensions, which means that there is no distinction between multiple uses of a metric throughout Istio.

For example, if we have a request_count metric that's tracked per service, the dimensions of the metric will include the service name such that there can be different counters, one per service. By not using the dimensions, the Circonus adapter is unable to distinguish all these uses and thus treats all request_count instances as the same thing.

Basically, the full identity of a metric is {metric_name, metric_dimensions}.

@redhotpenguin
